lib/puppet/resource.rb

```diff
@@ -11,9 +11,12 @@ class Puppet::Resource
   include Puppet::Util::Tagging
 
   include Enumerable
-  attr_accessor :file, :line, :catalog, :exported, :virtual, :validate_parameters, :strict
+  attr_accessor :file, :line, :catalog, :exported, :virtual, :strict
   attr_reader :type, :title
 
+  # @deprecated
```

Comment:
Kylo's previous comment on that attribute got lost during rebase. He wrote:

"FWIW I just grepped a recent drop (May 27th) of the forge and found no uses of `validate_parameters`.

I haven't grokked this entire PR, but if we're merely deprecating this attribute, but not removing it, that seems okay."